tracer/tracer.go

```diff
@@ -0,0 +1,126 @@
+package tracer
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"time"
+
+	"go.elastic.co/apm"
+	apmtransport "go.elastic.co/apm/transport"
+)
+
+type Tracer struct {
+	*apm.Tracer
+	logger  apm.Logger
+	timeout time.Duration
+}
+
+func NewTracer(logger apm.Logger, timeout time.Duration, serverSecret, serverUrl string) *Tracer {
+	tracer := apm.DefaultTracer
+	transport := tracer.Transport.(*apmtransport.HTTPTransport)
+	tracer.Transport = wrap(transport, logger, serverSecret, serverUrl)
```

Comment:
what is the custom transport providing now?

Reply:
right now nothing, because i realised that apm-server doesn't return accepted count on 202 requests.
but if we get that count, then we get the ratios of sent/accepted trough the custom transport. it would be cool then to compare ratios of dropped in agent / rejected in server and then come up with optimal configuration to minimize network usage and maximize useful agent work.

Reply:
Then I'd prefer to remove it until we need it.  Even when we do need it, I think we should consider using a custom transport in the `http.Client` instead of reimplementing the agent's higher level transport, if possible.